File: src/swift/exponentiationRecursive.swift
/* recursiveExponentiation.swift
Written by Roberto Martins on 10/10/2023
*/

func recursiveExponentiation<Number: Numeric>(base: Number, power: Int) -> Number {
    if (power == 0) {
        return 1
    }
    return base * recursiveExponentiation(base: base, power: power - 1)
}

func main() {
    print(recursiveExponentiation(base: 5, power: 3))
    print(recursiveExponentiation(base: 2.5, power: 4))
    print(recursiveExponentiation(base: 50, power: 0))
}

main()

File: src/c/ExponentiationRecursive.c
#include<stdio.h>



int exponenciacao(int base, int expoente){

	if(expoente==0) return 1;
	
	return (base * exponenciacao(base, expoente-1));

}



int main(){
	
	printf("%d\n", exponenciacao(5, 3));

	return 0;
}


//  In english 
// Function to calculate exponentiation using recursion
// int exponentiation(int base, int exponent) {
//     // Base case: any number raised to the power of 0 is 1
//     if (exponent == 0) return 1;
    
//     // Recursive case: multiply the base by the result of exponentiation with decreased exponent
//     return (base * exponentiation(base, exponent - 1));
// }

// int main() {
//     // Call the exponentiation function with base 5 and exponent 3, and print the result
//     printf("%d\n", exponentiation(5, 3));

//     return 0; // Exit the program
// }